{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -0,0 +1,141 @@
+"""
+Make model specifications
+
+click application that takes a template model specification file
+and locations for which to create model specs and uses jinja2 to
+render model specs with the correct location parameters plugged in.
+
+It will look for the model spec template in "model_spec.in" in the directory
+"../{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/model_specification".
+Specify multiple locations in a file called "locations.txt" in the same directory.
+You can create a single location using the "-s" option. Using the single
+location option will override an existing locations file.
+
+The application will look for the model spec and locations files based
+on the python environment that is active and these files don't need
+to be specified if the default names and location are used.
+"""
+import re
+import click
+from pathlib import Path
+from jinja2 import Template
+from collections import namedtuple
+from loguru import logger
+from typing import List, Callable
+from functools import wraps
+from sys import exit
+
+
+PROJ_NAME = '{{cookiecutter.package_name}}'
+Location = namedtuple('Location', ['loc_proper', 'loc_sanitized'])
+
+
+def sanitize(loc: str) -> (bool, Location):
+    loc_proper = loc.strip()
+    if loc_proper.startswith('#'):
+        loc_proper = loc_sanitized = ''
+    else:
+        loc_sanitized = loc_proper
+        loc_sanitized = re.sub("[- ,.&']", '_', loc_sanitized).lower()
+    return bool(loc_proper), Location(loc_proper, loc_sanitized)
+
+
+def get_sanitized_locations(single_loc: str, loc_file: Path) -> List[Location]:
+    """ Naming convention is to lowercase the location string, replace spaces with underscores,
+        and capitalize the first letter. No manipulation happens in the template code.
```

Comment:
Do we capitalize the first letter?  I think it should be all lower case, space turned to _, and apostrophes turned to -.

Reply:
We have 2 locations. The first, "location_proper", is not altered from how the user specified it. The second, "location_sanitized" is lowercased and any character in "[- ,.&']" is turned into an underscore. I looked through the names from the get_location_ids() to come up with the list of "oddball" characters. I can handle each one in a distinct way if necessary.

// specify "Russia"
    input_data:
        location: Russia
        input_draw_number: 0
        artifact_path: /share/costeffectiveness/artifacts/vtt/russia.hdf

    input_data:
        location: {% raw %}{{ location_proper }}{% endraw %}
        input_draw_number: 0
        artifact_path: /share/costeffectiveness/artifacts/{{cookiecutter.package_name}}/{% raw %}{{ location_sanitized }}{% endraw %}.hdf